### DISCUSSION TITLE
How to access a static uploadet file?

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1222] [Dev] [answered]
Hi
yet another develop question.
Imagine a user has uploaded a file.
Would it be possible to access this file in a processor somehow?
In the past, I used this to access a DB entry:
https://streampipes.apache.org/docs/docs/extend-sdk-static-properties.html#runtime-resolvable-selections
Is this still up to date or are some new methods implemented?

The workflow for the user would be something like:
1) Upload a file via file manager
2) Create a pipeline and select this specific processor
3) Select the file and maybe a property selection from the file and use it in the processor. Potentially add it to the stream as well.

**Comment by tenthe** [comment, marked as answer]
Hi @flomickl,

you can upload files as part of the processing element configuration. See `CsvMetadataEnrichment` as an example [1].
In the `declareModel` you can use `requiredFile`.
Is this the behavior you are looking for?


[1] https://github.com/apache/streampipes/blob/dev/streampipes-extensions/streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/csvmetadata/CsvMetadataEnrichmentController.java
